frontend_v2/src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -230,6 +246,22 @@ isSubmissionUsingUrl: any;
     this.authToken = this.globalService.getAuthToken();
   }
 
+  progress(files: FileList) {
+    this.isFileSelected = true;
+    const fileToUpload = files.item(0);
+    const seconds = fileToUpload.size;
```

Comment:
What's the unit of `fileToUpload.size`? Could this be taken equivalent to seconds?

Reply:
@lunayach what i am trying to do here is giving 10ms time for each byte of the filesize. As we are showing the progress bar as soon as we select the file so i couldn't find any way to get the status of file selection and so using this way to show the progress bar.

Reply:
How did you reach to the `10 ms` number?

Reply:
For this, i have considered the upload speed of 1Mbps

Reply:
https://usefulangle.com/post/67/pure-javascript-ajax-file-upload-showing-progess-percent
^ Can we have something like progress event of the `XMLHttpRequest.upload`?

Reply:
@lunayach This will require the File to be uploaded to the server and here we just need to show the progress bar on file selection. Please correct me if i missed something.

Reply:
I see. This was the deliverable of the issue #2116, `A progress bar which will appear on the upload of a file showing the percentage of the file uploaded to the server.` So, I think the progress bar should be showing the progress of the file uploaded. @RishabhJain2018 @Sanji515 please clarify if I misunderstood this.